### PR TITLE
Disable functions in readonly for inserts

### PR DESCRIPTION
**Pull request by SmitaRKulkarni**
Changelog category (leave one):
* Improvements

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Disable functions in readonly for inserts

Implementation:
* Added a new parameter to ITableFunction::execute to check if query is insert query.
* Used the new parameter in InterpreterInsertQuery::getTable function when calling ITableFunction::execute .

Testing:
* Added test case for INSERT in 02473_functions_in_readonly_mode.sh.

### CODE REVIEW

**Comment by nikitamikhaylov** [comment]
AST Fuzzer (debug) - OOM while executing this query `SELECT i FROM generateRandom('i Array(String)', NULL, 1048575) LIMIT 1048575` 
AST Fuzzer (tsan) - No server logs...

File: tests/queries/0_stateless/02473_functions_in_readonly_mode.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CURDIR"/../shell_config.sh
+
+$CLICKHOUSE_CLIENT --query="SELECT * from numbers(1)"
+$CLICKHOUSE_CLIENT --query="SELECT * from format('TSV', '123')"
+
+$CLICKHOUSE_CLIENT --readonly=1 --query="SELECT * from numbers(1)"
+$CLICKHOUSE_CLIENT --readonly=1 --query="SELECT * from format('TSV', '123')" 2>&1 | grep -Fq "Cannot execute query in readonly mode. (READONLY)" && echo 'ERROR' || echo 'OK'
```
Comment (nikitamikhaylov) [inline tests/queries/0_stateless/02473_functions_in_readonly_mode.sh:13]: @SmitaRKulkarni Why can't we execute this query in readonly mode?

Comment (SmitaRKulkarni) [reply tests/queries/0_stateless/02473_functions_in_readonly_mode.sh:13]: Only certain functions are allowed in readonly mode (`format` is not one of them) : https://github.com/ClickHouse/ClickHouse/pull/42708
This is an existing test. I used the same test file, changed it's extension to `.sh` and appended new insert tests

Comment (nikitamikhaylov) [reply tests/queries/0_stateless/02473_functions_in_readonly_mode.sh:13]: Okay

**Review by nikitamikhaylov — APPROVED** [review]

**Pull request merged**